### ISSUE TITLE
In-Game: #macro definitions do not support the inheritance implied by the IDE using a tree for the configs editor

### ISSUE THREAD

**Issue by Grisgram** [YoYoGames/GameMaker-Bugs#3935]
### Description
Imagine, I have set up my configs in the project like this (see attached image)

Now, from the hierarchy, one would expect, that "beta" is a child of "release".
I mean, in "release" ALL MACROS FROM DEFAULT ARE AVAILABLE, and I override only those, that shall be different from the default. So, I expected, the same is true between "release" and "beta".

I just tested this hierarchy with a macro:
```
#macro TESTVAL        42
#macro release:TESTVAL 1337

log($"********* {TESTVAL}");
```

Assuming, that "beta" derives from "release", I had expected to see the value 1337 in the log when starting in "beta" configuration.

But it printed 42.
So, even if I set up a hierarchy of configurations, they do not inherit.

thing is, in my current game i have a lot of macros defining room-order, available options, etc...
"default" is the dev-configuration.
without that inheritance this means to copy many macros with almost identical values from "release" to "beta" where only some change (this is why i did a hierarchy).
having all those settings multiple times opens up a new error source, if you forget to set one of those.
It also simply doubles the maintenance work if you need to adapt a setting.
Now imagine, it's not two, but many levels of hierarchy, not only release-beta, but release-beta-closedbeta-alpha or any set of possible build configurations. 
Copying all values x times can't be the way-to-go.

imho this should really be inherited, like in css or any other hierarchical system.

thanks for reading
cheers G

[4647ba86-90d2-4742-9074-94445b86693b](https://api.gamemaker.io/api/github/downloads/4647ba86-90d2-4742-9074-94445b86693b)

**Comment by rwkay** [comment]
Here is an example project I used while looking at this bug 
[MacroConfigHierarchy.zip](https://github.com/YoYoGames/GameMaker-Bugs/files/13938533/MacroConfigHierarchy.zip)

**Comment by rwkay** [comment]
Fixed in 2024.2

**Comment by Emc1923** [comment]
Verified as of  IDE v2024.200.0.478  Runtime v2024.200.0.493

**Comment by rwkay** [comment]
[Bug-3935-3-level-hierarchy-with-levels-missing.zip](https://github.com/YoYoGames/GameMaker-Bugs/files/14236699/Bug-3935-3-level-hierarchy-with-levels-missing.zip)
Reopened as bug found in a different example

**Comment by rwkay** [comment]
Fixed in 2024.2 

Fix issue with a hierarchy of configs when a macro definition is not present at each level

* Default -> test -> test_other - but no macro definition at the final one.
* Hierarchy was being traversed froom root -> leaf, it needs to be leaf -> root.

**Comment by mgeddesGM** [comment]
verified as of Beta IDE v2024.200.0.499  Runtime v2024.200.0.516
